fsharp-backend/src/LibExecution/TypeChecker.fs

```diff
@@ -85,6 +84,19 @@ let rec unify
   | TNull, DNull -> Ok()
   | TStr, DStr _ -> Ok()
   | TList _, DList _ -> Ok()
+  | TTuple (firstType, secondType, otherTypes), DTuple (first, second, theRest) ->
```

Comment:
It attempts to ensure args passed into fns match the expected types of the tuples.

Since we have no tuple fns to test with, and for now aren't supporting users to have tuple params (and return types) per https://github.com/darklang/dark/pull/4182#discussion_r912372713, I'm extracting type checking to https://github.com/darklang/dark/issues/4239